services/API-service/src/api/event/event.service.ts

```diff
@@ -1109,29 +1131,35 @@ export class EventService {
   // REFACTOR: this can be set up much better
   private async getEventEapAlertClass(
     disasterSettings: CountryDisasterSettingsEntity,
-    eventForecastSeverity: number,
-    eventForecastTrigger: boolean,
+    alertLevel: AlertLevel,
   ): Promise<EapAlertClass> {
     if (disasterSettings.disasterType === DisasterType.Floods) {
+      const alertLevelToEAPAlertClass = {
+        [AlertLevel.TRIGGER]: EapAlertClassKeyEnum.max,
+        [AlertLevel.WARNING]: EapAlertClassKeyEnum.med,
+        [AlertLevel.WARNINGMEDIUM]: EapAlertClassKeyEnum.med,
+        [AlertLevel.WARNINGLOW]: EapAlertClassKeyEnum.min,
+        [AlertLevel.NONE]: EapAlertClassKeyEnum.no,
+      };
       const eapAlertClasses = JSON.parse(
         JSON.stringify(disasterSettings.eapAlertClasses),
       );
       const alertClassKey = Object.keys(eapAlertClasses).find(
-        (key) => eapAlertClasses[key].value === eventForecastSeverity,
+        (key) => key === alertLevelToEAPAlertClass[alertLevel],
       );
       return {
         key: alertClassKey,
         ...eapAlertClasses[alertClassKey],
       };
     } else {
-      if (eventForecastTrigger) {
+      if (alertLevel === AlertLevel.TRIGGER) {
         return {
           key: EapAlertClassKeyEnum.max,
           label: 'Trigger',
           color: 'ibf-glofas-trigger',
           value: 1,
         };
-      } else if (eventForecastSeverity > 0) {
+      } else if (ALERT_LEVEL_WARNINGS.includes(alertLevel)) {
```

Comment:
I think this 'if' is particulary reserved for the general 'warning' class, and not for the low/medium warning classes? As I think in those cases they would need to be specified in country-config (like currently for floods).

Reply:
The code in this block uses code related to medium warnings so I'm not sure what you mean.

https://github.com/rodekruis/IBF-system/blob/6c490c442aef00ae742f315ca89f112e9410a22c/services/API-service/src/api/event/event.service.ts#L1163-L1168

Reply:
Yeah, I guess you're right. And either way, it will functionally work for now. It's just another reminder of how ugly this code setup around EapAlertClass is now. We already have AB#34026 for that.. but that doesn't mean much practically unless we can tie it to some feature/project item..

Reply:
@jannisvisser can we tie it to this feature/project and fix it after the core work is complete?

Reply:
I would be happy to if possible, but I guess you understand that this project's budget is likely largely depleted after the core work. Best we can do maybe is to tie it to QA work and do combined refactoring/docs/testing on this with our AA-day testing budget. I will at least drag it up a bunch for prioritization and add the refine tag.